text/0063-doc-comments.md

```diff
@@ -0,0 +1,226 @@
+
+## Detailed design
+
+Doc comments are designed to be very similar to annotations, and could reasonably be implemented as syntactic sugar for a special annotation annotation key.
+The primary difference is that, in order to fully satisfy the documentation use case, doc comments need to appear in more positions than are supported for annotations.
+We may choose to extend annotation to be supported in all positions proposed for doc comments, but this is not a necessary extensions to accept this RFC.
+
+Doc comments will be defined by this grammar.
+
+```
+DocCommentLine ::= '///' [^\r\n]* [\r\n]
+DocComment ::= {DocCommentLine}
+```
+
+### Policies
+
+Documentation for the policy as whole appears before the policy effect.
+It may be anywhere in relation to the annotations on that policy.
+Documentation for individual policy scope clauses must appear immediately before that clause.
+
+The policy grammar is now:
+
+```
+Policy ::= (Annotation|DocComment)} Effect '(' Scope ')' {Conditions} ';'
+Scope ::= [DocComment] Principal ',' [DocComment] Action ',' [DocComment] Resource
+```
+
+### Schema
+
+Documentation for a declaration in a schema appears immediately before the `entity`, `action`, `type`, or `namespace` keyword at the start of the declaration.
+The fields of the attributes in a record type can also be documented with the comment appearing directly before the attribute name.
+A record type might be written for an entities attributes, actions context, or common type definition, or it might be nested inside of some other type.
+Doc comments are supported in all of these locations.
+
+The schema grammar is now:
+
+```
+Namespace := ([DocComment] 'namespace' Path '{' {Decl} '}') | Decl
+Decl := [DocComment] (Entity | Action | TypeDecl)
+AttrDecls := [DocComment] Name ['?'] ':' Type [',' | ',' AttrDecls]
+```
+
+### Improperly placed comments
+
+Comments starting with the `///` that are not in a legal positions for doc comments are an error.
+The definition of comments is accordingly updated to
+
+```
+COMMENT ::= '//' ([^/\r\n] [^\r\n]*)? [\r\n]
+```
+
+This makes the RFC a breaking change for any users who wrote comments in this style, but I believe that is preferable to silently ignoring what a user intended to be a doc comment.
+For an initial and backwards compatible change, improperly place doc comments could be allowed with a warning.
+We can then make this an error in a future major version of the Cedar, or never make it an error if the break is not acceptable.
+
+### Alternative doc comment syntax
+
+Choosing to use Rust style doc comments feels natural after developing Cedar in Rust, but we should consider alternative syntax options.
+Some options include using block style doc comments (`/** .. */`), using another common comment character for doc comments (e.g., `#`), and picking a different third character following hte usual two slashes (e.g., `//@` to emphasize the connection with annotations).
```

Comment:
I agree that we should consider all of these options; I'd like to hear others' thoughts on this.

Reply:
@philhassey suggests just using `//` as in Godoc which I hadn't considered as an option. Not a huge fan, but it's worth considering.

Reply:
No strong feelings on my end. I like the look of `///`, but likely I'm biased from having mostly used Rust lately. My second preference would be `//@` (or some other third character like `!` or `#`).